m2cgen/assemblers/utils.py

```diff
@@ -57,3 +57,26 @@ def to_2d_array(var):
     else:
         x, y = 1, np.size(var)
     return np.reshape(np.asarray(var), (x, y))
+
+
+def sigmoid_expr(expr):
+    neg_expr = ast.BinNumExpr(ast.NumVal(0), expr, ast.BinNumOpType.SUB)
+    exp_expr = ast.ExpExpr(neg_expr)
+    return ast.BinNumExpr(
+        ast.NumVal(1),
+        ast.BinNumExpr(ast.NumVal(1), exp_expr, ast.BinNumOpType.ADD),
+        ast.BinNumOpType.DIV,
+        to_reuse=True)
+
+
+def softmax_exprs(exprs):
+    exp_exprs = [ast.ExpExpr(e, to_reuse=True) for e in exprs]
```

Comment:
I'm not sure I understand why we need `to_reuse` for these expressions as they seem to only used once.

Reply:
Nope. The used twice: once to calculate some and the second time - for normalization (`exp(val[i]) / sum`